mindocr/data/transforms/east_transforms.py

```diff
@@ -0,0 +1,429 @@
+      scale   : feature map / image
+      length  : image length
+    Output:
+      score gt, geo gt, ignored
+    """
+    score_map = np.zeros(
+        (int(img.height * scale), int(img.width * scale), 1), np.float32)
+    geo_map = np.zeros(
+        (int(img.height * scale), int(img.width * scale), 5), np.float32)
+    ignored_map = np.zeros(
+        (int(img.height * scale), int(img.width * scale), 1), np.float32)
+
+    index = np.arange(0, length, int(1 / scale))
+    index_x, index_y = np.meshgrid(index, index)
+    ignored_polys = []
+    polys = []
+
+    for i, vertice in enumerate(vertices):
+        if labels[i] == 0:
+            ignored_polys.append(np.around(scale * vertice.reshape((4, 2))).astype(np.int32))
+            continue
+
+        poly = np.around(scale * shrink_poly(vertice).reshape((4, 2))).astype(np.int32)
+        polys.append(poly)
+        temp_mask = np.zeros(score_map.shape[:-1], np.float32)
+        cv2.fillPoly(temp_mask, [poly], 1)
+
+        theta = find_min_rect_angle(vertice)
+        rotate_mat = get_rotate_mat(theta)
+
+        rotated_vertices = rotate_vertices(vertice, theta)
+        x_min, x_max, y_min, y_max = get_boundary(rotated_vertices)
+        rotated_x, rotated_y = rotate_all_pixels(rotate_mat, vertice[0], vertice[1], length)
+
+        d1 = rotated_y - y_min
+        d1[d1 < 0] = 0
+        d2 = y_max - rotated_y
+        d2[d2 < 0] = 0
+        d3 = rotated_x - x_min
+        d3[d3 < 0] = 0
+        d4 = x_max - rotated_x
+        d4[d4 < 0] = 0
+        geo_map[:, :, 0] += d1[index_y, index_x] * temp_mask
+        geo_map[:, :, 1] += d2[index_y, index_x] * temp_mask
+        geo_map[:, :, 2] += d3[index_y, index_x] * temp_mask
+        geo_map[:, :, 3] += d4[index_y, index_x] * temp_mask
+        geo_map[:, :, 4] += theta * temp_mask
+
+    cv2.fillPoly(ignored_map, ignored_polys, 1)
+    cv2.fillPoly(score_map, polys, 1)
+    return score_map, geo_map, ignored_map
+
+
+def extract_vertices(data_labels):
+    """
+    extract vertices info from txt lines
+    Input:
+      lines   : list of string info
+    Output:
+      vertices: vertices of text regions <numpy.ndarray, (n,8)>
+      labels  : 1->valid, 0->ignore, <numpy.ndarray, (n,)>
+    """
+    vertices_list = []
+    labels_list = []
+    data_labels = eval(data_labels)
+    for data_label in data_labels:
+        vertices = data_label['points']
+        vertices = [item for point in vertices for item in point]
+        vertices_list.append(vertices)
+        labels = 0 if data_label['transcription'] == '###' else 1
+        labels_list.append(labels)
+    return np.array(vertices_list), np.array(labels_list)
+
+
+class EASTProcessTrain:
+    def __init__(self, scale=0.25, length=512):
+        super(EASTProcessTrain, self).__init__()
+        self.scale = scale
+        self.length = length
+
+    def __call__(self, data):
+        vertices, labels = extract_vertices(data['label'])
+        img = Image.open(data['img_path'])
```

Comment:
Why is the image loaded here? Shouldn't it be inside data loader?

Reply:
Modified, using data['image'] from dataloader instead.